### DISCUSSION TITLE
Ability to test permissions right in the modal

### DISCUSSION THREAD

**Discussion by navanshu** [directus/directus#17721] [Feature Requests]
### Summary

Permission are a very important part of the whole system. 
It gets confusing on how to configure the data. Also there is the thing that docs mentions that O2M grants permission to the child object. I think it will provide a robust and complete environment

### Basic Example

It would be great if we had a mini panel right at the bottom of the modal where one can try out different users based on roles and see why was the permission granted. 

### Motivation

The docs part of the permission is not that good and I feel like it would make working with permissions very easy and reliable. As the permissions part is quite difficult as a low code tool.

### Detailed Design

Even if the permissions modal provide a bit more information about the data types on the left side and the right side data can also  provided with a picker, It will make the whole thing more robust. Maybe the modal can show the code implementation of the logic that is result of the selection made by user and on hover it can show the data type of the variables.

### Must Have:
- A A mini panel to run data to see if the permission are correct
- B Right side data picker component

### Should Have:
- C Hover to check the data types

### Could Have:
- D Presenting the permission as code

### Won't Have:
- E Edit permission as code

### Drawbacks

A test panel or a select component for dynamic variable would be great and will also be useful in flows.
These two feature will make working with directus easier. 

I feel like data types can be quite complex as it would require knowledge of the JavaScript compiler itself.
But it provides a type safe environment like typescript.

### Alternatives

I feel like it can also be done with post man.
But you are not 100% sure that it was correct.

### Adoption Strategy

I don't think it will cause a breaking change

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
